### DISCUSSION TITLE
Is it possible to mint some other kind of NFTs?

### DISCUSSION THREAD

**Discussion by zed-wong** [MixinNetwork/mixin#112] [Q&A] [answered]
Is it possible to mint NFTs with image provided by users? I've tried to modify `collection` argument mentioned in [here](https://github.com/MixinNetwork/nfo/blob/4e12fb92c7105b005e0b48cb8f0665c899ca7cb5/README.md), but I still got the same kind of image in wallet. What's required to do to mint some other kind of NFTs?

**Comment by zed-wong** [comment]
I found my question might not be clear enough.

When using [GET /collectibles/tokens/UUID](https://developers.mixin.one/docs/api/collectibles/outputs#get-collectiblestokensuuid), we can get 

```"meta":{"group":"Bar","name":"Foo","description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. ","icon_url":"https://mixin-images.zeromesh.net/zVDjOxNTQvVsA8h2B4ZVxuHoCF3DJszufYKWpd9duXUSbSapoZadC7_13cnWBqg0EmwmRcKGbJaUpA8wFfpgZA=s128","media_url":"https://mixin-images.zeromesh.net/HvYGJsV5TGeZ-X9Ek3FEQohQZ3fE9LBEBGcOcn4c4BNHovP4fW4YB97Dg5LcXoQ1hUjMEgjbl1DPlKg1TW7kK6XP=s128","mime":"image/png","hash":"2edae50309823f0cf8aed5a3d6f08499900eeaf711b3e9a093dd5234ce528b9a"}, }``` in return. 

But I couldn't find how those data are generated. And how can I modify those data?

There are three arguments required to call `nft.BuildMintNFO` mentioned in [here](https://github.com/MixinNetwork/nfo/blob/main/README.md#mint-nft). 

In [nfo/messenger.go](https://github.com/MixinNetwork/nfo/blob/main/messenger.go#L120):
> 'collection' is [`uuid.Nil.String()`](https://github.com/MixinNetwork/nfo/blob/main/messenger.go#L120) ('00000000-0000-0000-0000-000000000000')
> 'id' is [`rw.generateTokenId(msgId)`](https://github.com/MixinNetwork/nfo/blob/main/messenger.go#L115) (looks like some random generated unique id)
> 'hash' is [`crypto.NewHash([]byte("TEST:" + msgId))`](https://github.com/MixinNetwork/nfo/blob/main/messenger.go#L119) (also looks like nothing)

I've tried to modify `collection` into '11111111-0000-0000-0000-000000000000' and everything goes well as usual.

So I'm quite confused with how to modify the meta data? And where are those data come from?

**Comment by cedricfung** [comment]
It's possible to do that. But it all depends on the wallet specification. Now Mixin Messenger will try to fetch those data from the NFT marketplace @an-lee is building. I guess he is able to provide some guides and updates.

**Comment by an-lee** [comment, marked as answer]
I wrote a guide about it: https://docs.thetrident.one/api/guide/how-to-mint-nft

**Comment by zed-wong** [comment]
Thanks guys.

**Comment by cedricfung** [comment]
Now I see @an-lee has made his market testing publicly https://trident-test.onrender.com
